File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Base.ViewInfo;
using DevExpress.Utils;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraGrid.Views.Grid;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable("Parent");
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("ID", typeof(int));
            tbl.Columns.Add("Number", typeof(int));
            tbl.Columns.Add("Date", typeof(DateTime));
            tbl.Columns.Add("DETAILID", typeof(int));
            for (int i = 0; i < RowCount; i++)
                tbl.Rows.Add(new object[] { String.Format("Name{0}", i), i, 3 - i, DateTime.Now.AddDays(i), i });
            return tbl;
        }

        private DataTable CreateDet1Table(int RowCount)
        {
            DataTable tbl = new DataTable("Det1");
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("ID", typeof(int));
            for (int j = 0; j < RowCount; j++)
                for (int i = 0; i < RowCount; i++)
                    tbl.Rows.Add(new object[] { String.Format("Detail1Name{0}", i), i });
            return tbl;
        }

        private DataTable CreateDet2Table(int RowCount)
        {
            DataTable tbl = new DataTable("Det2");
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("ID", typeof(int));
            for (int j = 0; j < RowCount; j++)
                for (int i = 0; i < RowCount; i++)
                    tbl.Rows.Add(new object[] { String.Format("Detail2Name{0}", i), i });
            return tbl;
        }


        private DataSet GetMasterDetail()
        {
            DataSet ds = new DataSet("TestDS");
            ds.Tables.Add(CreateTable(20));
            ds.Tables.Add(CreateDet1Table(20));
            ds.Tables.Add(CreateDet2Table(20));
            DataColumn parentColumn = ds.Tables["Parent"].Columns["DETAILID"];
            DataColumn childColumn = ds.Tables["Det1"].Columns["ID"];
            ds.Relations.Add(new DataRelation("relDet1", parentColumn, childColumn));
            childColumn = ds.Tables["Det2"].Columns["ID"];
            ds.Relations.Add(new DataRelation("relDet2", parentColumn, childColumn));
            return ds;
        }


        public Form1()
        {
            InitializeComponent();
            gridControl1.DataSource = GetMasterDetail();
            gridControl1.DataMember = "Parent";
        }

        public SuperToolTip GetToolTipByHitInfo(GridView view, GridHitInfo hi)
        {
            if (hi == null)
                return null;
            SuperToolTip toolTip = new SuperToolTip();
            string caption = view.GridControl.MainView.Equals(view) ? "Main View" : string.Format("DetailView: {0}", view.LevelName);
            toolTip.Items.AddTitle(caption);
            toolTip.Items.AddSeparator();
            if (!hi.InRowCell)
                return null;
            toolTip.Items.Add(String.Format("Cell at {0}:", hi.HitPoint));
            toolTip.Items.Add(string.Format("Column = {0}", hi.Column));
            toolTip.Items.Add(string.Format("RowHandle = {0}", hi.RowHandle));
            return toolTip;
        }




        private void defaultToolTipController1_DefaultController_GetActiveObjectInfo(object sender, DevExpress.Utils.ToolTipControllerGetActiveObjectInfoEventArgs e)
        {
            GridControl gridControl = e.SelectedControl as GridControl; 
            if (gridControl == null)
                return;
            GridView view = gridControl.GetViewAt(e.ControlMousePosition) as GridView;
            if (view == null)
                return;
            GridHitInfo hi = view.CalcHitInfo(e.ControlMousePosition);
            SuperToolTip toolTip = GetToolTipByHitInfo(view, hi);
            if (toolTip == null)
                return;
            e.Info = new ToolTipControlInfo();
            e.Info.Object = hi.Column;
            e.SelectedObject = hi.Column;
            e.Info.SuperTip = toolTip;
            e.Info.ToolTipType = ToolTipType.SuperTip;
            e.Info.ToolTipPosition = Control.MousePosition;
            e.Info.ImmediateToolTip = true;
        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Base.ViewInfo
Imports DevExpress.Utils
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.XtraGrid.Views.Grid

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
		Private Function CreateTable(ByVal RowCount As Integer) As DataTable
			Dim tbl As New DataTable("Parent")
			tbl.Columns.Add("Name", GetType(String))
			tbl.Columns.Add("ID", GetType(Integer))
			tbl.Columns.Add("Number", GetType(Integer))
			tbl.Columns.Add("Date", GetType(DateTime))
			tbl.Columns.Add("DETAILID", GetType(Integer))
			For i As Integer = 0 To RowCount - 1
				tbl.Rows.Add(New Object() { String.Format("Name{0}", i), i, 3 - i, DateTime.Now.AddDays(i), i })
			Next i
			Return tbl
		End Function

		Private Function CreateDet1Table(ByVal RowCount As Integer) As DataTable
			Dim tbl As New DataTable("Det1")
			tbl.Columns.Add("Name", GetType(String))
			tbl.Columns.Add("ID", GetType(Integer))
			For j As Integer = 0 To RowCount - 1
				For i As Integer = 0 To RowCount - 1
					tbl.Rows.Add(New Object() { String.Format("Detail1Name{0}", i), i })
				Next i
			Next j
			Return tbl
		End Function

		Private Function CreateDet2Table(ByVal RowCount As Integer) As DataTable
			Dim tbl As New DataTable("Det2")
			tbl.Columns.Add("Name", GetType(String))
			tbl.Columns.Add("ID", GetType(Integer))
			For j As Integer = 0 To RowCount - 1
				For i As Integer = 0 To RowCount - 1
					tbl.Rows.Add(New Object() { String.Format("Detail2Name{0}", i), i })
				Next i
			Next j
			Return tbl
		End Function


		Private Function GetMasterDetail() As DataSet
			Dim ds As New DataSet("TestDS")
			ds.Tables.Add(CreateTable(20))
			ds.Tables.Add(CreateDet1Table(20))
			ds.Tables.Add(CreateDet2Table(20))
			Dim parentColumn As DataColumn = ds.Tables("Parent").Columns("DETAILID")
			Dim childColumn As DataColumn = ds.Tables("Det1").Columns("ID")
			ds.Relations.Add(New DataRelation("relDet1", parentColumn, childColumn))
			childColumn = ds.Tables("Det2").Columns("ID")
			ds.Relations.Add(New DataRelation("relDet2", parentColumn, childColumn))
			Return ds
		End Function


		Public Sub New()
			InitializeComponent()
			gridControl1.DataSource = GetMasterDetail()
			gridControl1.DataMember = "Parent"
		End Sub

		Public Function GetToolTipByHitInfo(ByVal view As GridView, ByVal hi As GridHitInfo) As SuperToolTip
			If hi Is Nothing Then
				Return Nothing
			End If
			Dim toolTip As New SuperToolTip()
			Dim caption As String
			If view.GridControl.MainView.Equals(view) Then
				caption = "Main View"
			Else
				caption = String.Format("DetailView: {0}", view.LevelName)
			End If
			toolTip.Items.AddTitle(caption)
			toolTip.Items.AddSeparator()
			If (Not hi.InRowCell) Then
				Return Nothing
			End If
			toolTip.Items.Add(String.Format("Cell at {0}:", hi.HitPoint))
			toolTip.Items.Add(String.Format("Column = {0}", hi.Column))
			toolTip.Items.Add(String.Format("RowHandle = {0}", hi.RowHandle))
			Return toolTip
		End Function




		Private Sub defaultToolTipController1_DefaultController_GetActiveObjectInfo(ByVal sender As Object, ByVal e As DevExpress.Utils.ToolTipControllerGetActiveObjectInfoEventArgs) Handles defaultToolTipController1.DefaultController.GetActiveObjectInfo
			Dim gridControl As GridControl = TryCast(e.SelectedControl, GridControl)
			If gridControl Is Nothing Then
				Return
			End If
			Dim view As GridView = TryCast(gridControl.GetViewAt(e.ControlMousePosition), GridView)
			If view Is Nothing Then
				Return
			End If
			Dim hi As GridHitInfo = view.CalcHitInfo(e.ControlMousePosition)
			Dim toolTip As SuperToolTip = GetToolTipByHitInfo(view, hi)
			If toolTip Is Nothing Then
				Return
			End If
			e.Info = New ToolTipControlInfo()
			e.Info.Object = hi.Column
			e.SelectedObject = hi.Column
			e.Info.SuperTip = toolTip
			e.Info.ToolTipType = ToolTipType.SuperTip
			e.Info.ToolTipPosition = Control.MousePosition
			e.Info.ImmediateToolTip = True
		End Sub
	End Class
End Namespace